cf/mixin/fielddomain.py

```diff
@@ -626,49 +699,214 @@ def _point_not_in_cell(nodes_x, nodes_y, point):
             if ind is not None:
                 mask_component_shape = []
                 masked_subspace_size = 1
-                ind = np.array(ind)
+                # TODONUMPY2: https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword
+                ind = np.array(ind, copy=False)
 
                 for i, (axis, start, stop) in enumerate(
                     zip(canonical_axes, ind.min(axis=1), ind.max(axis=1))
                 ):
                     if data_axes and axis not in data_axes:
                         continue
 
-                    if indices[axis] == slice(None):
-                        if compress:
-                            # Create a compressed index for this axis
-                            size = stop - start + 1
-                            index = sorted(set(ind[i]))
-                        elif envelope:
-                            # Create an envelope index for this axis
-                            stop += 1
-                            size = stop - start
-                            index = slice(start, stop)
-                        elif full:
-                            # Create a full index for this axis
-                            start = 0
-                            stop = domain_axes[axis].get_size()
-                            size = stop - start
-                            index = slice(None)
-                        else:
-                            raise ValueError(
-                                "Must have mode full, envelope or compress"
-                            )  # pragma: no cover
+                    if compress:
+                        # Create a compressed index for this axis
+                        size = stop - start + 1
+                        index = sorted(set(ind[i]))
+                    elif envelope:
+                        # Create an envelope index for this axis
+                        stop += 1
+                        size = stop - start
+                        index = slice(start, stop)
+                    elif full:
+                        # Create a full index for this axis
+                        start = 0
+                        stop = domain_axes[axis].get_size()
+                        size = stop - start
+                        index = slice(None)
+                    else:
+                        raise ValueError(
+                            "Must have mode full, envelope, or compress"
+                        )  # pragma: no cover
+
+                    # Overwrite the placeholder value of None
+                    if indices[axis] is not None:
+                        raise ValueError(
+                            "This error means that there is a bug: The "
+                            "'indices' dictionary should contain None for "
+                            "each axis with an 'ind'."
+                        )
```

Comment:
The trouble with `assert` is that the user can turn them off, so I don't like them in library code.

Reply:
OK, fair enough. I have noticed we don't use them in the NCAS cf* codebases which made me suspect you had your reasons against them. And happy to go with the 'house rules'!

I do doubt that most (except certain CMS/power) users would report anything if they reached this error, though of course we hope they never reach it at all. Maybe we could add a bit to ask them to let us know if they see it, e.g. add "If you see this, please raise an issue at <Issue Tracker URL>"? Or is that overkill?

Reply:
Maybe overkill - it was more a note to me and you :)

Reply:
Well, that's exactly why I think `assert`s are great for that exact purpose - for the developers rather than the users. But I won't push for us to use one here or elsewhere :slightly_smiling_face: